executor/executor.go

```diff
@@ -1111,6 +1118,101 @@ func (e *AggregationExec) innerNext() (ret bool, err error) {
 	return true, nil
 }
 
+// StreamAggExec deals with all the aggregate functions.
+// It is built from Aggregate Plan. When Next() is called, it reads all the data from Src and updates all the items in AggFuncs.
+type StreamAggExec struct {
+	Src          Executor
+	schema       expression.Schema
+	ResultFields []*ast.ResultField
+	executed     bool
+	mode         plan.AggregationType
+	ctx          context.Context
+	AggFuncs     []expression.AggregationFunction
+	GroupByItems []expression.Expression
+}
+
+// Close implements Executor Close interface.
+func (e *StreamAggExec) Close() error {
+	e.executed = false
+	for _, agg := range e.AggFuncs {
+		agg.Clear()
+	}
+	return e.Src.Close()
+}
+
+// Schema implements Executor Schema interface.
+func (e *StreamAggExec) Schema() expression.Schema {
+	return e.schema
+}
+
+// Fields implements Executor Fields interface.
+func (e *StreamAggExec) Fields() []*ast.ResultField {
+	return e.ResultFields
+}
+
+// Next implements Executor Next interface.
+func (e *StreamAggExec) Next() (*Row, error) {
+	if e.executed {
+		return nil, nil
+	}
+	var groupKey []byte
+	for {
+		row, err := e.Src.Next()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if row == nil {
+			e.executed = true
+			break
+		}
+		groupKey, err = e.getGroupKey(row)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		var end bool
+		for _, af := range e.AggFuncs {
+			end, err = af.StreamUpdate(row.Data, groupKey, e.ctx)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+		if end {
+			break
+		}
+	}
+	retRow := &Row{Data: make([]types.Datum, 0, len(e.AggFuncs))}
```

Comment:
If there is groupby items and you do not get any row from e.src. Will this Next() return a row?